pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -386,6 +386,15 @@ public static class Server {
         "pinot.server.starter.enableRealtimeOffsetBasedConsumptionStatusChecker";
     public static final boolean DEFAULT_ENABLE_REALTIME_OFFSET_BASED_CONSUMPTION_STATUS_CHECKER = false;
 
+    public static final String CONFIG_OF_ENABLE_REALTIME_FRESHNESS_BASED_CONSUMPTION_STATUS_CHECKER =
```

Comment:
can we ensure that exactly one of the config is set (maybe in the starter), and throw exception if not?

Reply:
I had it this way in a previous iteration but went back for a few reasons:
- @navina suggested this approach
- the behaviors here aren't orthogonal. The freshness checker is strictly more robust than the offset checker. If you enable both, getting the more robust one is perfectly acceptable from an operational standpoint (in my opinion)
- both of these are disabled by default. anyone running pinot today is at most using the offset based checker. so they would be taking a tangible action to enable the freshness checker meaning it's what they wanted enabled anyway. Having the server fail at startup because of this seems like poor UX

Reply:
What happens if/when someone comes up with yet another way of doing this?

Reply:
If someone wants an orthogonal way to do this, then it's worth further discussion there. The previous code was already doing something like this:
- if nothing is set, just start
- if realtime wait is set, wait a static amount of time before starting
- if offset is set, do nothing
- if offset and realtime wait is set, do the offset checker

At this point we've somewhat exhausted the "am I caught up" methods with either freshness or offset based. If someone eventually wants something more complex like "freshness and offset based together", we can migrate this config to dynamically point to the class name of the offset checker they want.